src/widgets/dialogs/qcolordialog.cpp

```diff
@@ -1571,8 +1572,30 @@ bool QColorDialogPrivate::selectColor(const QColor &col)
 
 QColor QColorDialogPrivate::grabScreenColor(const QPoint &p)
 {
-    const QDesktopWidget *desktop = QApplication::desktop();
-    const QPixmap pixmap = QGuiApplication::primaryScreen()->grabWindow(desktop->winId(), p.x(), p.y(), 1, 1);
+    // Get the screen at p if it exists.
+    QScreen *screen = QGuiApplication::screenAt(p);
+    QScreen *primary_screen = QGuiApplication::primaryScreen();
```

Comment:
Perhaps use camel case for 'primary_screen' and also for 'screen_p' below to be in keeping with the convention used.